src/contract-read.ts

```diff
@@ -7,6 +7,12 @@ import GQLResultInterface, { GQLEdgeInterface, GQLTransactionsResultInterface }
 
 import SmartWeaveError, { SmartWeaveErrorType } from './errors';
 
+const cache: {
```

Comment:
how long does this cache live in the memory? I'm asking because of current issues with memory leaks ( https://discord.com/channels/357957786904166400/756557551234973696/857784751149940736 ) - I wonder if it doesn't introduce the risk of another leak...

Reply:
@ppedziwiatr The cache lives in memory until the function is done running.

Reply:
As discussed here: https://discord.com/channels/357957786904166400/756557551234973696/860136978380095509 - it is a globally scoped cache and lives in the memory after this function exits - just leaving for future reference.